src/hera/workflows/script.py

```diff
@@ -760,6 +755,27 @@ def script_wrapper(func: Callable[FuncIns, FuncR]) -> Callable:
         def task_wrapper(*args, **kwargs) -> Union[FuncR, Step, Task, None]:
             """Invokes a `Script` object's `__call__` method using the given SubNode (Step or Task) args/kwargs."""
             if _context.active:
+                if len(args) == 1 and isinstance(args[0], (InputV1, InputV2)):
```

Comment:
These looks like it doesn't support kwargs.

```suggestion
                if (
                    (len(args) == 1 and len(kwargs) == 0 and isinstance(args[0], (InputV1, InputV2))) or
                    (len(args) == 0 and len(kwargs) == 1 and isinstance(next(iter(kwargs.values())), (InputV1, InputV2)))
                ):
                    if args:
                        arguments = args[0]._get_as_arguments()
                    else:
                        arguments = next(iter(kwargs.values()))._get_as_arguments()

```

Reply:
Agreed. I copied that limitation from the existing Pydantic I/O logic here: https://github.com/argoproj-labs/hera/blob/2eec4c282557823c661dae6b410b470d13a90be5/src/hera/workflows/_meta_mixins.py#L530-L532

I think if we want to lift that limitation and support the user passing their Input type as a named parameter, we should do it in both places at once in a follow-up PR.

Reply:
sounds great!

Reply:
>looks like it doesn't support kwargs

This is by design as we need a way to separate the function input arg from the Task kwargs, i.e. the `Input` should be positional-only so we don't have a kwarg name clash